File: JavaScript/896. Monotonic Array.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var isMonotonic = function(nums) {
    //nums.sort((a, b) => {return a - b;});
    var a = 0;
    for(var i = 0;i < nums.length-1;i++){
        if(!(nums[i] <= nums[i+1])){
            a += 1;
            break;
        }
    }
    for(var i = 0;i < nums.length-1;i++){
        if(!(nums[i] >= nums[i+1])){
            a += 1;
            break;
        }
    }
    if(a == 2) return false;
    else return true;
};

File: C++/896. Monotonic Array.cpp
class Solution {
public:
    bool isMonotonic(vector<int>& nums) {
        ios_base::sync_with_stdio(false);
        cin.tie(NULL);
        cout.tie(NULL);

        int l = nums.size();

        bool increase = nums[0] < nums[l - 1];

        for (int i = 1; i < nums.size(); i++) {
            if (increase && nums[i - 1] > nums[i]) return false;
            else if (!increase && nums[i - 1] < nums[i]) return false; 
        }

        return true;
    }
};
